### PR TITLE
Retry downloading the benchmarks datasets

### PR DESCRIPTION
**Pull request by irevoire**
Downloading the benchmarks datasets is failing [more and more](https://github.com/meilisearch/milli/pull/607#pullrequestreview-1076023074) often; thus, instead of fixing the issue, I thought we could retry multiple times.

### CODE REVIEW

File: benchmarks/build.rs [hunk 1]
```diff
@@ -89,6 +89,15 @@ fn main() -> anyhow::Result<()> {
     Ok(())
 }
 
+fn retry<Ok, Err>(fun: impl Fn() -> Result<Ok, Err>, times: usize) -> Result<Ok, Err> {
+    for _ in 0..times {
+        if let ok @ Ok(_) = fun() {
+            return ok;
+        }
+    }
```
Comment (Kerollmops) [inline benchmarks/build.rs:97]: Don't we want to sleep, here?

**Review by Kerollmops — CHANGES_REQUESTED** [review]

File: benchmarks/build.rs (hunk 1 above)
Comment (irevoire) [reply benchmarks/build.rs:97]: As you want, I don't kwow why it’s failing

Comment (Kerollmops) [reply benchmarks/build.rs:97]: I neither know 😂

**Comment by irevoire** [comment]
Soooo... Should we merge, or should I sleep?
I would say we can merge it like that, and if it doesn't solve anything, then we add a sleep in it 👀

**Review by Kerollmops — APPROVED** [review]
Indeed, let's merge and see if the problem persists!
bors merge

**Pull request merged**